src/main/scala/tech/cryptonomic/conseil/tezos/michelson/parser/JsonParser.scala

```diff
@@ -120,25 +98,51 @@ object JsonParser {
 
   case class JsonSimpleInstruction(
       prim: String,
-      args: Option[List[Either[JsonExpression, List[JsonInstruction]]]] = None,
+      args: Option[List[EmbeddedElement]] = None,
       annots: Option[List[String]] = None
   ) extends JsonInstruction {
     override def toMichelsonInstruction =
       MichelsonSingleInstruction(
         name = prim,
         annotations = annots.getOrElse(List.empty),
-        embeddedElements = args.getOrElse(List.empty).map {
-          case Left(jsonExpression) => jsonExpression.toMichelsonExpression
-          case Right(Nil) => MichelsonEmptyInstruction
-          case Right(jsonInstructions) => MichelsonInstructionSequence(jsonInstructions.map(_.toMichelsonInstruction))
-        }
+        embeddedElements = args.getOrElse(List.empty).map(toMichelsonElement)
       )
   }
 
   case class JsonInstructionSequence(instructions: List[JsonInstruction]) extends JsonInstruction {
     override def toMichelsonInstruction = MichelsonInstructionSequence(instructions.map(_.toMichelsonInstruction))
   }
 
+  /*
+   * Wrapper for int constant
+   *
+   * {"int": "0"}
+   *
+   * */
+  case class JsonIntConstant(int: String) extends JsonInstruction {
+    override def toMichelsonInstruction = MichelsonIntConstant(int.toLong)
+  }
+
+  /*
+   * Wrapper for string constant
+   *
+   * {"string": "0"}
+   *
+   * */
+  case class JsonStringConstant(string: String) extends JsonInstruction {
+    override def toMichelsonInstruction = MichelsonStringConstant(string)
+  }
+
+  /*
+   * Wrapper for bytes constant
+   *
+   * {"bytes": "0500"}
+   *
+   * */
+  case class JsonBytesConstant(bytes: String) extends JsonInstruction {
+    override def toMichelsonInstruction = MichelsonBytesConstant(bytes)
```

Comment:
Just to clarify, does this value encode byte-strings? That is, multiple byte-encoded values?